openwisp_radius/api/views.py

```diff
@@ -377,17 +373,53 @@ def dispatch(self, *args, **kwargs):
         return super().dispatch(*args, **kwargs)
 
     def validate_membership(self, user):
-        if (self.organization.pk,) not in user.organizations_pk:
+        if not (user.is_superuser or (self.organization.pk,) in user.organizations_pk):
             message = _('User "{}" is not member ' 'of "{}"').format(
                 user.username, self.organization.slug
             )
             logger.warning(message)
             raise serializers.ValidationError({'non_field_errors': [message]})
 
 
+class IsAllowedRadiusBatch(BasePermission):
+    def has_permission(self, request, view):
+        radbatch_model_name = swapper.get_model_name('openwisp_radius', 'RadiusBatch')
+        app_label, _ = swapper.split(radbatch_model_name)
+        return bool(
+            request.user
+            and (request.user.is_superuser or request.user.is_staff)
+            and request.user.has_perm(f'{app_label}.view_radiusbatch')
+        )
+
+
+class DownloadRadiusBatchPdfView(DispatchOrgMixin, APIView):
+    authentication_classes = (UserTokenAuthentication, SessionAuthentication)
+    permission_classes = (IsAllowedRadiusBatch,)
```

Comment:
I think we could use `('IsAdminUser'', 'DjangoModelPermissions')`, (having to define IsAllowedRadiusBatch should not be needed), you will need to specify the `queryset` attribute (eg: `RadiusBatch.objects.all()`).

Reply:
Thanks for this, much better! :smile: